test/Constraints/diagnostics.swift

```diff
@@ -930,6 +930,38 @@ let _ = (r29850459() ? r29850459_a : r29850459_b) + 42.0 // expected-error {{bin
 let _ = ((r29850459_flag || r29850459()) ? r29850459_a : r29850459_b) + 42.0 // expected-error {{binary operator '+' cannot be applied to operands of type 'Int' and 'Double'}}
 // expected-note@-1 {{overloads for '+' exist with these partially matching parameter lists: (Double, Double), (Int, Int), (Int, UnsafeMutablePointer<Pointee>), (Int, UnsafePointer<Pointee>)}}
 
+// SR-6272: Tailored diagnostics with fixits for numerical conversions
+
+func SR_6272_a() {
+  enum Foo: Int {
+    case bar
+  }
+  // expected-error@+2 {{binary operator '*' cannot be applied to operands of type 'Int' and 'Float'}} {{22-28=}} {{29-30=}}
+  // expected-note@+1 {{overloads for '*' exist with these partially matching parameter lists: (Float, Float), (Int, Int)}}
+  Foo.bar.rawValue * Float(0)
```

Comment:
How does this fare with a contextual type?

```swift
let x: Float = Foo.bar.rawValue * Float(0)
```

Reply:
Good point.. 
It suggests to fix `Float(0)` to `0` so far..

Reply:
`tryIntegerCastFixIts` treats the `toType` as a contextual type, always.  You may want to try calling it on the LHS and RHS expressions of the binary operation and pass along their respective types and the contextual type.

Reply:
Addressed with https://github.com/apple/swift/pull/13272/commits/72cf1585a00c7b2c29d3cad5ae6d06b19d6ff692.